packages/phalanx/test/Kokkos/tKokkosViewOfViews.cpp

```diff
@@ -0,0 +1,60 @@
+#include "Kokkos_Core.hpp"
+#include "Teuchos_UnitTestHarness.hpp"
+
+// This test demonstrates how to create a view of views for double and FAD types.
+using exec_t = Kokkos::DefaultExecutionSpace;
+using mem_t = Kokkos::DefaultExecutionSpace::memory_space;
+
+TEUCHOS_UNIT_TEST(ViewOfViews,double) {
+
+  const int num_cells = 10;
+  const int num_pts = 8;
+  const int num_equations = 32;
+
+  Kokkos::View<double***,mem_t> a("a",num_cells,num_pts,num_equations);
+  Kokkos::View<double***,mem_t> b("b",num_cells,num_pts,num_equations);
+  Kokkos::View<double***,mem_t> c("c",num_cells,num_pts,num_equations);
+
+  Kokkos::deep_copy(a,2.0);
+  Kokkos::deep_copy(b,3.0);
+
+  // Requirement 1: The inner view must be unmanaged to prevent double deleteion!
+  using InnerView = Kokkos::View<double***,mem_t,Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+  using OuterView = Kokkos::View<InnerView*,mem_t>;
+
+  // Requirement 2: The host view must exist for the life of the device view!
+  OuterView::HostMirror v_host("outer host",3);
+  {
+    v_host(0) = a;
+    v_host(1) = b;
+    v_host(2) = c;
+
+    OuterView v_dev("outer device",3);
+    Kokkos::deep_copy(v_dev,v_host);
+
+    auto policy = Kokkos::MDRangePolicy<Kokkos::Rank<3>>({0,0,0},{num_cells,num_pts,num_equations});
+    Kokkos::parallel_for("view of view test",policy,KOKKOS_LAMBDA (const int cell,const int pt, const int eq) {
+      v_dev(2)(cell,pt,eq) = v_dev(0)(cell,pt,eq) + v_dev(1)(cell,pt,eq);
+    });
+
+    auto c_host = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(),c);
+    Kokkos::deep_copy(c_host,c);
+
```

Comment:
already copied above

Reply:
isn't line 40 a copy of c->c_host??

Reply:
oops. I thought you were pointing to the other deep copy a few lines above. I'll fix that. Thanks!